server/src/main/java/com/google/fhir/gateway/AllowedQueriesConfig.java

```diff
@@ -27,17 +28,25 @@
 class AllowedQueriesConfig {
   public static final String MATCHES_ANY_VALUE = "ANY_VALUE";
 
+  public static Pattern MATCHES_PATH_ANY_VALUE_PATTERN = Pattern.compile("^[a-zA-Z]*(/ANY_VALUE)$");
```

Comment:
How have you decided what characters to allow in the initial segment of the path (i.e., `[a-zA-Z]*`)? For example why not allow numbers or `_` or anything else that is allowed in URL paths? If the goal is to limit this to only FHIR resources, then shouldn't we make it more specific?

Reply:
Here I thought of limiting this to just FHIR resources. and also to <Resource>/ANY_VALUE

A generic path can even be like Resource1/<id>/Resource2/<id>/ANY_VALUE but this is not a valid FHIR path.
If we have to keep this generic then it is better to just support a regex and not have any restrictions. 
I didn't do this because we discussed to just support the minimal set of requirements and later expand that if required.

Reply:
We should clarify what is the contract we are supporting here. Right now it is a mix of the following two:
1) we only let a FHIR resource to be the prefix OR
2) we allow any prefix
I am okay with leaning toward the first option (FHIR resources only) but that is not what we are enforcing. So I would suggest let's go with the second option, i.e, any prefix is allowed. So a request goes through if the prefix part matches exactly, regardless of its content. If you feel enforcing FHIR resources is important for a particular use-case, please comment here.

Reply:
sg. updated to allow any prefix